src/handler/mod.rs

```diff
@@ -0,0 +1,111 @@
+/// This module contains most of the low-level commands
+/// and grub variable modifications
+use anyhow::{bail, Result};
+
+use std::process::Command;
+use std::str;
+
+/// reboots the system if boot_counter is greater than 0 or can be forced too
+pub fn handle_reboot(force: bool) -> Result<()> {
+    if !force {
+        if let Some(t) = get_boot_counter()? {
+            if t <= 0 {
+                bail!("countdown ended, check greenboot-rollback status")
+            };
+        }
+    }
+    log::info!("restarting the system");
+    Command::new("systemctl").arg("reboot").status()?;
+    Ok(())
+}
+
+/// rollback to previous ostree deployment if boot counter is less than 0
+pub fn handle_rollback() -> Result<()> {
+    if let Some(boot_counter) = get_boot_counter()? {
+        if boot_counter <= 0 {
+            log::info!("Greenboot will now attempt to rollback");
+            Command::new("rpm-ostree").arg("rollback").status()?;
+            return Ok(());
+        }
+    }
+    bail!("Rollback not initiated");
+}
+
+/// sets grub variable boot_counter if not set
+pub fn set_boot_counter(reboot_count: i32) -> Result<()> {
+    match get_boot_counter() {
+        Ok(Some(current_counter)) => {
+            log::info!("boot_counter={current_counter}");
+            bail!("counter already set");
+        }
+        _ => {
+            //will still try to set boot_counter to override cases like boot_counter=<some_string>
+            log::info!("setting boot counter");
+            set_grub_var("boot_counter", reboot_count)?;
+        }
+    }
+    Ok(())
+}
+
+/// resets grub variable boot_counter
+pub fn unset_boot_counter() -> Result<()> {
+    Command::new("grub2-editenv")
+        .arg("-")
+        .arg("unset")
+        .arg("boot_counter")
+        .status()?;
+    Ok(())
+}
+
+/// sets grub variable boot_success
+pub fn handle_boot_status(success: bool) -> Result<()> {
```

Comment:
nit: I would rename this function to `set_boot_success` or something similar, it does not handle things, it sets a value